src/plan_expand_hypertable.c

```diff
@@ -51,54 +114,73 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 		{
 			Node	   *qual = (Node *) lfirst(lc);
 			RestrictInfo *restrictinfo;
+			Relids		relids;
+			bool		belongs_to_cur_rel = false;
+
+			if (IsA(qual, BoolExpr))
```

Comment:
This should be removed now, since the function will error when not at top level. This also doesnt detect all non-toplevel uses.

Reply:
I think it's better to catch things early on, even if it does not cover all the possible problems.

Reply:
Well either you check this properly which means using expression_tree_walker or you rely on the fact that this will error anyway for nested expressions because the function itself throws an error. Considering that this is a critical path i think we should do the latter.

Reply:
Here we only test for nesting `chunks_in` function within boolean expressions. Note that even if you use function on the top level but combine with OR, the query tree will have it nested under boolean. Since this function is used in WHERE clause I anticipate that combining it with boolean logic (ANDing only is not a problem) might be a common mistake and this code should prevent that.